polling/poller.py

```diff
@@ -0,0 +1,104 @@
+"""
+Skeleton program which continually polls for new dispatches
+"""
+
+import os
+import urllib
+import datetime
+import time
+from typing import Dict
+
+import requests
+from dateutil import parser as date_parser
+
+
+VOLTUS_API_URL = os.getenv("VOLTUS_API_URL", "https://api.voltus.co")
+VOLTUS_API_KEY = os.getenv("VOLTUS_API_KEY", "secret")
+POLLING_INTERVAL_SECONDS = 30
+
+s = requests.Session()
+s.headers.update({"X-Voltus-API-Key": VOLTUS_API_KEY, "Accept": "application/json"})
+managed_dispatches = {}
+cancelled_dispatches = set()
+
+
+def create_dispatch(dispatch_info: Dict):
+    # TODO: Fill in with your company-specific logic
+    #       Remember that this could get called twice if the service gets restarted
+    #       Remember that end_time could be None
+    #       sites could have one sites, many sites, or all of your sites
+    print(
+        "Dispatch with ID {} starting at {} and ending at {} for sites {} created".format(
+            dispatch_info["id"],
+            dispatch_info["start_time"],
+            dispatch_info.get("end_time", "not specified"),
+            dispatch_info["sites"],
+        )
+    )
+
+
+def update_dispatch(dispatch_info: Dict):
+    # TODO: Fill in with your company-specific logic
+    #       start_time and end_time can change to be earlier or later
+    #       Remember that this could get called twice if the service is restarted
+    print(
+        "Dispatch with ID {} starting at {} and ending at {} for sites {} updated".format(
+            dispatch_info["id"],
+            dispatch_info["start_time"],
+            dispatch_info.get("end_time", "not specified"),
+            dispatch_info["sites"],
+        )
+    )
+
+
+def cancel_dispatch(dispatch_info: Dict):
+    # TODO: Fill in with your company-specific logic to cancel this dispatch, likely by ID
+    #       Remember that this could get called twice if the service is restarted
+    print("Dispatch with ID {} cancelled".format(dispatch_info["id"]))
+
+
```

Comment:
maybe add a comment here to say this starts the service? not sure what is idiomatic python

Reply:
`if __name__ == "__main__":` is actually the usual idiom here, which has the side benefit of allowing this code to be imported as a module.